cogs/Music.py

```diff
@@ -0,0 +1,360 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Dec 19 2020
+
+@author: sebas
+"""
+
+import discord
+import asyncio
+from async_timeout import timeout
+from discord.ext import commands
+from discord.ext import menus
+from discord.ext.commands import BucketType, cooldown, CommandOnCooldown
+import youtube_dl
+import time
+
+# Suppress noise about console usage from errors
+youtube_dl.utils.bug_reports_message = lambda: ''
+
+ytdl_format_options = {
+    'format': 'bestaudio/best',
+    'outtmpl': '%(extractor)s-%(id)s-%(title)s.%(ext)s',
+    'restrictfilenames': True,
+    'noplaylist': True,
+    'nocheckcertificate': True,
+    'ignoreerrors': False,
+    'logtostderr': False,
+    'quiet': True,
+    'no_warnings': True,
+    'default_search': 'auto',
+    'source_address': '0.0.0.0' # bind to ipv4 since ipv6 addresses cause issues sometimes
+}
+
+ffmpeg_options = {
+    'options': '-vn'
+}
+
+ytdl = youtube_dl.YoutubeDL(ytdl_format_options)
+
+dashes = []
+for i in range(0,20):
+    dashes.append("".join(["▬"]*i))
+
+class YTDLSource(discord.PCMVolumeTransformer): #taken from examples
+    def __init__(self, source, *, data, volume=0.5):

```

Comment:
what is data exactly is it the data for the video on youtube

Reply:
When you download something from ytdl it will include a dictionary of the video's information eg channel. length, etc.